### PR TITLE
Add style for proc calling with simple blocks

### PR DESCRIPTION
**Pull request by NewAlexandria**
Add pass-as-argument style for proc invocation in simple blocks.

* [x] Removed the unrelated `block` guidance
* [x] \(hopefully) clarified guidance wording
* [x] checked to be sure that the 'bad' pattern was not used where in the guide, too

[See the discussion here](https://github.com/bbatsov/ruby-style-guide/issues/419)

### CODE REVIEW

**Comment by NewAlexandria** [comment]
Right, so awkwardly enough, if you're doing a puts then you do need an each:

```Ruby
names.each { |name| puts name }
```

because `puts` is not a method of any element in the `names` array.  This only works when invoking the same method (or returning the same property) of each object in the array / collection.

File: README.md [hunk 1]
```diff
@@ -1124,13 +1124,25 @@ condition](#safe-assignment-in-condition).
...
+  ```
 
+* <a name="block-chaining">Chaining</a>
+
+  ```Ruby
   # good
-  names.select { |name| name.start_with?('S') }.map { |name| name.upcase }
+  names.select { |name| name.start_with?("S") }.map(&:upcase)
+
+  # bad
```
Comment (bbatsov) [inline README.md]: Bad? This unrelated change has no place in this PR.

File: README.md [hunk 2]
```diff
@@ -1124,13 +1124,25 @@ condition](#safe-assignment-in-condition).
   # good
   names.each { |name| puts name }
 
+  Proc invokation shorthand when that is the only operation of a block
```
Comment (bbatsov) [inline README.md]: This is pretty unclear. I'm OK with have a rule suggesting the use of &:something when possible, but it should have crystal clear wording and a separate bullet point.

**Comment by NewAlexandria** [comment]
Pardon, it looks like I was really distracted and didn't re-review before opening.

Made changes and updated the PR description.

**Comment by NewAlexandria** [comment]
I simplified this

**Comment by NewAlexandria** [comment]
bump

File: README.md [hunk 3]
```diff
@@ -1124,13 +1124,25 @@ condition](#safe-assignment-in-condition).
...
+  Use this proc invocation shorthand when the invoked method / attr is the only operation of a block
+
+  # bad
+  names.map { |name| name.upcase }
+
+  # good
+  names.map(&:upcase)
+  ```
+
+* <a name="block-chaining">Chaining</a>
```
Comment (bbatsov) [inline README.md]: I'm still confused by the change you made here. Using the shorthand syntax should have been a separate rule.

**Comment by bbatsov** [comment]
@NewAlexandria ping

**Comment by NewAlexandria** [comment]
I moved the added bit into its own rule, above.  I kept the change to the block-chaining example, which still reads well, linearly, because this rule exists above.

File: README.md [hunk 4]
```diff
@@ -1106,6 +1106,19 @@ condition](#safe-assignment-in-condition).
...
+  Use the proc invocation shorthand when the invoked method / attr is the only operation of a block.
+<sup>[[link](#single-action-blocks)]</sup>
+
+  ```Ruby
+  # bad
+  names.map { |name| name.upcase }
+
+  # good
+  names.map(&:upcase)
+
```
Comment (bbatsov) [inline README.md]: Remove this blank line.

File: README.md [hunk 5]
```diff
@@ -1106,6 +1106,19 @@ condition](#safe-assignment-in-condition).
   'test'.upcase
   ```
 
+* <a name="single-action-blocks"></a>
+  Use the proc invocation shorthand when the invoked method / attr is the only operation of a block.
```
Comment (bbatsov) [inline README.md]: Remove `attr` - there are only methods, nothing more.

File: README.md [hunk 6]
```diff
@@ -1124,13 +1137,14 @@ condition](#safe-assignment-in-condition).
   # good
   names.each { |name| puts name }
 
+
```
Comment (bbatsov) [inline README.md]: Remove this blank.

File: README.md [hunk 7]
```diff
@@ -1124,13 +1137,14 @@ condition](#safe-assignment-in-condition).
...
   # bad
   names.select do |name|
     name.start_with?('S')
   end.map { |name| name.upcase }
 
   # good
-  names.select { |name| name.start_with?('S') }.map { |name| name.upcase }
+  names.select { |name| name.start_with?("S") }.map(&:upcase)
```
Comment (bbatsov) [inline README.md]: Use single quotes around `S`.

File: README.md [hunk 8]
```diff
@@ -1106,6 +1106,18 @@ condition](#safe-assignment-in-condition).
   'test'.upcase
   ```
 
+* <a name="single-action-blocks"></a>
+  Use the proc invocation shorthand when the invoked method the only operation of a block.
```
Comment (bbatsov) [inline README.md]: method is the only
